ts/component/table-builder.ts

```diff
@@ -708,29 +718,33 @@ export class TableBuilder<ParentType, ChildType> {
         );
       });
       // create the second row for child headers
-      const childRow = this.thead.insertRow();
+      const childRow = addHeaderRow(false);
+      if (this.definition.bulkActions) {
+        this.addSelectAllHeader(childRow);
+      }
       this.columns.forEach((column, i) => {
+        const combinedClass = column.headingClass
+          ? `text-black ${column.headingClass}`
+          : "text-black";
         addColumnHeader(
           childRow,
           column.title,
           i === 0 && !this.definition.bulkActions,
-          "text-black",
+          combinedClass,
           "bg-grey-100"
         );
       });
     } else {
-      // only one row of headers (normal case)
-      const row = this.thead.insertRow();
+      // single row of headers
+      const row = addHeaderRow(false);
       if (this.definition.bulkActions) {
         this.addSelectAllHeader(row);
-      } else {
-        this.selectAllCheckbox = undefined;
       }
       this.columns.forEach((column, i) => {
         addColumnHeader(
           row,
           column.title,
-          i == 0 && !this.definition.bulkActions,
+          i === 0 && !this.definition.bulkActions,
           column.headingClass
         );
```

Comment:
This is all duplicated. The only difference is the last two parameters of `addColumnHeader`